src/main/scala/shine/C/TranslationContext.scala

```diff
@@ -15,9 +15,12 @@ class TranslationContext() extends shine.DPIA.Compilation.TranslationContext {
 
       case _: IndexType => Assign(dt, lhs, rhs)
 
-      //TODO think about this more thoroughly
-      case PairType(_, _) => Assign(dt, lhs, rhs)
-        // assign(dt1, recordAcc1(dt1, dt2, lhs), fst(rhs)) `;` assign(dt2, recordAcc2(dt1, dt2, lhs), snd(rhs))
+      // FIXME: both solutions currently create issues
+      // TODO: think about this more thoroughly
+      // case PairType(_, _) => Assign(dt, lhs, rhs)
+      case PairType(dt1, dt2) =>
+        assign(dt1, pairAcc1(dt1, dt2, lhs), fst(rhs)) `;`
+        assign(dt2, pairAcc2(dt1, dt2, lhs), snd(rhs))
```

Comment:
I plan to open an issue about this part.

Reply:
yes, please.

Reply:
https://github.com/rise-lang/shine/issues/58 is opened